sounddevice.py

```diff
@@ -2349,6 +2376,97 @@ def __init__(self, channel_selectors):
+            This sets MacCore's sample rate conversion quality.
+            Possible values are: ``'Min'``, ``'Low'``, ``'Medium'``,
+            ``'High'``, or ``'Max'``.  ``'Max'`` is the default.
+
+        Example
+        -------
+        Input from 'Mic 9' and 'Mic 10' on RME Fireface UFX+, and send
+        output to 'Analog 3' and 'Analog 4':
+
+        >>> import sounddevice as sd
+        >>>
+        >>> input_channels = [8, 9]
+        >>> output_channels = [2, 3]
+        >>>
+        >>> # For input devices, the channel_map is simply the list of
+        >>> # 0-based index numbers for each channel:
+        >>> ch_map_in = input_channels
+        >>> mc_in = sd.MacCoreSettings(channel_map=ch_map_in)
+        >>>
+        >>> # For output devices, the list needs to be "inverted":
+        >>> # Populate the array with -1 for unwanted channels:
+        >>> max_output_channels = query_devices(device, 'output')['max_output_channels']
+        >>> ch_map_out = [-1 for _ in range(max_output_channels)]
+        >>> # For this example device, max_output_channels == 94.
+        >>> # Now overwrite any -1's for channels we'll use for output:
+        >>> for ndx in range(len(output_channels)):
+        >>>     ch_map_out[output_channels[ndx]]=ndx
+        >>> mc_out = sd.MacCoreSettings(channel_map=ch_map_out)
+        >>>
+        >>> sd.playrec(..., channels=2, extra_settings=(mc_in, mc_out))
+        """
+        # Check inputs:
+        conversion_dict = {
+            'Min':    _lib.paMacCoreConversionQualityMin   ,
+            'Low':    _lib.paMacCoreConversionQualityLow   ,
+            'Medium': _lib.paMacCoreConversionQualityMedium,
+            'High':   _lib.paMacCoreConversionQualityHigh  ,
+            'Max':    _lib.paMacCoreConversionQualityMax   ,
+            }
+        if rate_conv_quality not in conversion_dict:
+            raise ValueError('rate_conv_quality must be one of {!r}'.format(list(conversion_dict.keys())))
+
+        if channel_map is not None:
+            if not all(isinstance(ch, int) for ch in channel_map):
+                raise TypeError('channel_map must be an iterable (such as list or tuple) of int')
+
+        self._flags = conversion_dict[rate_conv_quality]
+        if change_dev_params:
+            self._flags |= _lib.paMacCoreChangeDeviceParameters
+        if prevent_rate_conv:
+            self._flags |= _lib.paMacCoreFailIfConversionRequired
+
+        # this struct must be kept alive!
+        self._streaminfo = _ffi.new('PaMacCoreStreamInfo*')
+        _lib.PaMacCore_SetupStreamInfo(self._streaminfo, self._flags)
+
+        if channel_map is not None:
+            # this array must be kept alive!
+            self._channel_map = _ffi.new('SInt32[]', channel_map)
+            _lib.PaMacCore_SetupChannelMap(self._streaminfo,
+                                           self._channel_map,
+                                           len(channel_map))
```

Comment:
If `channel_map` is *just* an iterable, you can't get it's length here.
You can use `len(self._channel_map)` though.